isomorphe/geonetwork.py

```diff
@@ -343,6 +364,86 @@ def get_groups(self) -> dict:
         return r.json()
 
 
+class GeonetworkClientV3(GeonetworkClient):
+    version = 3
+
+    def _search_params(self, query: dict[str, Any] | None) -> dict[str, Any]:
+        params = {
+            "_content_type": "json",
+            "buildSummary": "false",
+            "fast": "index",  # needed to get info such as title
+            "sortBy": "changeDate",
+            "sortOrder": "reverse",
+        }
+        if query:
+            params |= query
+        return params
+
+    def _search_hits(self, params: dict[str, Any], from_pos: int) -> list[dict]:
+        r = self.session.get(
+            f"{self.api}/q",
+            headers={"Accept": "application/json"},
+            params=params | {"from": from_pos},
+        )
+        r.raise_for_status()
+        rsp = r.json()
+        hits = rsp.get("metadata")
+        if hits and "geonet:info" in hits:
+            # When returning a single record, metadata isn't a list :/
+            hits = [hits]
+        return hits
+
+    def _as_record(self, hit: dict) -> Record:
+        return Record(
+            uuid=hit["geonet:info"]["uuid"],
+            title=hit.get("defaultTitle"),
```

Comment:
Typehinting is complaining here

```suggestion
            title=hit.get("defaultTitle", ""),
```